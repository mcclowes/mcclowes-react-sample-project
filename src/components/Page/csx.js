import styled from "styled-components/macro";

export const PageWrapper = styled("div")`
  align-items: center;
  background: ${(props) => props.theme.colors.primary};
  background: linear-gradient(
    52deg,
    ${(props) => props.theme.colors.primary} 0%,
    ${(props) => props.theme.colors.alt} 100%
  );
  display: flex;
  height: 100vh;
  justify-content: center;
  overflow: hidden;
  width: 100vw;
`;

export const PageContainer = styled("div")`
  background: ${(props) => props.theme.colors.white};
  border-radius: ${(props) => props.theme.sizes.border.radius};
  display: flex;
  flex-direction: column;
  padding: 1em 2em;
  text-align: center;

  -webkit-box-shadow: 0px 4px 28px 6px rgba(0, 0, 0, 0.3);
  -moz-box-shadow: 0px 4px 28px 6px rgba(0, 0, 0, 0.3);
  box-shadow: 0px 4px 28px 6px rgba(0, 0, 0, 0.3);
`;
