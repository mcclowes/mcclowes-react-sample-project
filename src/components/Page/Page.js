import Anime from "react-anime";
import React from "react";
import { PageContainer, PageWrapper } from "./csx";

const Page = (props) => {
  return (
    <PageWrapper>
      <Anime
        opacity={[0, 1]}
        translateY={["-4em", 0]}
        delay={(e, i) => i * 2000}
      >
        <PageContainer>{props.children}</PageContainer>
      </Anime>
    </PageWrapper>
  );
};

export default Page;
