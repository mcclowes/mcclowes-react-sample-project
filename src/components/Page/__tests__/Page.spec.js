import Page from "../Page";
import React from "react";
import { shallow } from "enzyme";

const DEFAULT_PROPS = {};

const renderComponent = (newProps) => {
  const props = {
    ...DEFAULT_PROPS,
    ...newProps,
  };

  return shallow(<Page {...props} />);
};

let wrapper;

describe("Page", () => {
  it("renders component", () => {
    wrapper = renderComponent();

    expect(wrapper).toMatchSnapshot();
  });
});
