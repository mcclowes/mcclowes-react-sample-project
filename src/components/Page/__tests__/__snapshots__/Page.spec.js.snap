// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders component 1`] = `
<csx__PageWrapper>
  <e
    delay={[Function]}
    opacity={
      Array [
        0,
        1,
      ]
    }
    translateY={
      Array [
        "-4em",
        0,
      ]
    }
  >
    <csx__PageContainer />
  </e>
</csx__PageWrapper>
`;
